src/lib/transaction_snark/test/zkapps_examples/big_circuit/big_circuit.ml

```diff
@@ -0,0 +1,142 @@
+      ; account = Accept
+      }
+  ; call_type = Call
+  ; use_full_commitment = true
+  ; authorization_kind = Signature
+  }
+
+let deploy_account_update : Account_update.t =
+  { body = deploy_account_update_body
+  ; authorization = Signature Signature.dummy
+  }
+
+let account_updates =
+  []
+  |> Zkapp_command.Call_forest.cons_tree account_update
+  |> Zkapp_command.Call_forest.cons deploy_account_update
+
+let memo = Signed_command_memo.empty
+
+let transaction_commitment : Zkapp_command.Transaction_commitment.t =
+  let account_updates_hash = Zkapp_command.Call_forest.hash account_updates in
+  Zkapp_command.Transaction_commitment.create ~account_updates_hash
+
+let fee_payer =
+  { Account_update.Fee_payer.body =
+      { Account_update.Body.Fee_payer.dummy with
+        public_key = pk_compressed
+      ; fee = Currency.Fee.(of_nanomina_int_exn 100)
+      }
+  ; authorization = Signature.dummy
+  }
+
+let full_commitment =
+  Zkapp_command.Transaction_commitment.create_complete transaction_commitment
+    ~memo_hash:(Signed_command_memo.hash memo)
+    ~fee_payer_hash:
+      (Zkapp_command.Digest.Account_update.create
+         (Account_update.of_fee_payer fee_payer) )
+
+let sign_all ({ fee_payer; account_updates; memo } : Zkapp_command.t) :
+    Zkapp_command.t =
+  let fee_payer =
+    match fee_payer with
+    | { body = { public_key; _ }; _ }
+      when Public_key.Compressed.equal public_key pk_compressed ->
+        { fee_payer with
+          authorization =
+            Schnorr.Chunked.sign sk
+              (Random_oracle.Input.Chunked.field full_commitment)
+        }
+    | fee_payer ->
+        fee_payer
+  in
+  let account_updates =
+    Zkapp_command.Call_forest.map account_updates ~f:(function
+      | ({ body = { public_key; use_full_commitment; _ }
+         ; authorization = Signature _
+         } as account_update :
+          Account_update.t )
+        when Public_key.Compressed.equal public_key pk_compressed ->
+          let commitment =
+            if use_full_commitment then full_commitment
+            else transaction_commitment
+          in
+          { account_update with
+            authorization =
+              Control.Signature
+                (Schnorr.Chunked.sign sk
+                   (Random_oracle.Input.Chunked.field commitment) )
+          }
+      | account_update ->
+          account_update )
+  in
+  { fee_payer; account_updates; memo }
+
+let zkapp_command : Zkapp_command.t =
+  sign_all
+    { fee_payer = { body = fee_payer.body; authorization = Signature.dummy }
+    ; account_updates
+    ; memo
+    }
+
+let () =
+  Ledger.with_ledger ~depth:ledger_depth ~f:(fun ledger ->
+      let (_ : _) =
+        Ledger.get_or_create_account ledger account_id
+          (Account.create account_id
+             Currency.Balance.(
+               Option.value_exn
+                 (add_amount zero (Currency.Amount.of_nanomina_int_exn 500))) )
+      in
+      let thing _ =
```

Comment:
Inline in a thunk, rather than calling it thing?

Reply:
the sad thing is that I thought about it, but then thought about CI restarting : D